src/components/Api.js

```diff
@@ -0,0 +1,104 @@
+export default class Api {
+  constructor({ baseURL, headers }) {
+    this._baseURL = baseURL;
+    this._headers = headers;
+  }
+
+  _checkPromise(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    // if the server returns an error, reject the promise
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseURL}/cards`, {
+      headers: this._headers,
+    }).then((res) => {
+      return this._checkPromise(res);
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkPromise)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.